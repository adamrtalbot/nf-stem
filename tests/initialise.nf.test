nextflow_pipeline {

    name "Test INITIALISE"
    script "subworkflows/local/initialise.nf"

    test("Should run without failures") {

        then {
            assert workflow.success
            assert snapshot(workflow).match()
        }

    }

    test("Should show help") {
        when {
            params {
                help = true
            }
        }

        then {
            assert workflow.success
            assert snapshot(workflow).match()
        }

    }

    test("Should show version") {
        when {
            params {
                version = true
            }
        }

        then {
            assert workflow.success
            assert snapshot(workflow).match()
        }

    }

    test("Should fail because no params.input") {
        when {
            params {
                input = null
            }
        }

        then {
            assert workflow.failed
            assert workflow.exitStatus == 1
            assert snapshot(workflow).match()
        }

    }

}
