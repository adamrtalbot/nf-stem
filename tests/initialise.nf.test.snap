{
    "Should show help": {
        "content": [
            {
                "stderr": [
                    
                ],
                "errorReport": "",
                "exitStatus": 0,
                "out": {
                    
                },
                "failed": false,
                "stdout": [
                    
                ],
                "errorMessage": "",
                "trace": {
                    "tasksFailed": 0,
                    "tasksCount": 0,
                    "tasksSucceeded": 0
                },
                "name": "workflow",
                "success": true
            }
        ],
        "timestamp": "2023-07-26T13:34:15+0000"
    },
    "Should fail because no params.input": {
        "content": [
            {
                "stderr": [
                    
                ],
                "errorReport": "",
                "exitStatus": 1,
                "out": {
                    
                },
                "failed": true,
                "stdout": [
                    "Please provide an input samplesheet to the pipeline e.g. '--input samplesheet.csv'"
                ],
                "errorMessage": "",
                "trace": {
                    "tasksFailed": 0,
                    "tasksCount": 0,
                    "tasksSucceeded": 0
                },
                "name": "workflow",
                "success": false
            }
        ],
        "timestamp": "2023-07-26T13:34:15+0000"
    },
    "Should run without failures": {
        "content": [
            {
                "stderr": [
                    
                ],
                "errorReport": "",
                "exitStatus": 0,
                "out": {
                    
                },
                "failed": false,
                "stdout": [
                    
                ],
                "errorMessage": "",
                "trace": {
                    "tasksFailed": 0,
                    "tasksCount": 0,
                    "tasksSucceeded": 0
                },
                "name": "workflow",
                "success": true
            }
        ],
        "timestamp": "2023-07-26T13:34:15+0000"
    },
    "Should show version": {
        "content": [
            {
                "stderr": [
                    
                ],
                "errorReport": "",
                "exitStatus": 0,
                "out": {
                    
                },
                "failed": false,
                "stdout": [
                    
                ],
                "errorMessage": "",
                "trace": {
                    "tasksFailed": 0,
                    "tasksCount": 0,
                    "tasksSucceeded": 0
                },
                "name": "workflow",
                "success": true
            }
        ],
        "timestamp": "2023-07-26T13:34:15+0000"
    }
}